Fix domain validation method

DL6ER opened the pull request:
**By submitting this pull request, I confirm the following:** 

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md), as well as this entire template.
- [X] I have made only one major change in my proposed changes.
- [X] I have commented my proposed changes within the code.
- [X] I have tested my proposed changes, and have included unit tests where possible.
- [X] I am willing to help maintain this change if there are issues with it later.
- [X] I give this submission freely and claim no ownership.
- [X] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [X] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

---
**What does this PR aim to accomplish?:**

Fix https://github.com/pi-hole/AdminLTE/issues/1848

**How does this PR accomplish the above?:**

Move validation functions into func.php and use the same validation in all cases. Interestingly enough, the code used in many other places has been modified to allow underscores almost exactly 4 years ago (https://github.com/pi-hole/AdminLTE/pull/533). We keep this now and make it the routine used everywhere.

**What documentation changes (if any) are needed to support this PR?:**

None

DL6ER committed:
Move validation functions into func.php and use the same validation in all cases.

Signed-off-by: DL6ER <dl6er@dl6er.de>

DL6ER commented:
Rebased on latest `release/v5.6`

PromoFaux commented:
![image](https://user-images.githubusercontent.com/1998970/126904141-402d3c6f-cc19-4cbb-8a6b-8ea3ddfe9fb3.png)

DL6ER committed:
Fix inverse logic when adding domains on the group management pages.

Signed-off-by: DL6ER <dl6er@dl6er.de>

DL6ER commented:
Thanks, fixed. I also added reasons for why adding a domain failed. Example:

![Screenshot from 2021-07-25 19-02-30](https://user-images.githubusercontent.com/16748619/126907308-d0d7eaa7-2645-4e44-b883-feb9468639bc.png)
